test/sslprovidertest.c

```diff
@@ -7,32 +7,49 @@
  * https://www.openssl.org/source/license.html
  */
 
+#include <string.h>
 #include <openssl/provider.h>
 
 #include "ssltestlib.h"
 #include "testutil.h"
 
 static char *cert = NULL;
 static char *privkey = NULL;
+static char *modulename = NULL;
+static char *configfile = NULL;
 
 /* TODO(3.0): Re-enable this code. See comment in setup_tests() */
-#if 0
 OSSL_PROVIDER *defctxlegacy = NULL;
-#endif
 
 static int test_different_libctx(void)
 {
     SSL_CTX *cctx = NULL, *sctx = NULL;
     SSL *clientssl = NULL, *serverssl = NULL;
     int testresult = 0;
     OPENSSL_CTX *libctx = OPENSSL_CTX_new();
+    OSSL_PROVIDER *prov = NULL;
 
-/* TODO(3.0): Re-enable this code. See comment in setup_tests() */
-#if 0
-    /* Verify that the default provider in the default libctx is not available */
-    if (!TEST_false(OSSL_PROVIDER_available(NULL, "default")))
+    /*
+     * Verify that the default and fips providers in the default libctx iare not
+     * available
+     */
+    if (!TEST_false(OSSL_PROVIDER_available(NULL, "default"))
+            || !TEST_false(OSSL_PROVIDER_available(NULL, "fips")))
+        goto end;
+
+    if (!TEST_true(OPENSSL_CTX_load_config(libctx, configfile)))
         goto end;
-#endif
+
+    prov = OSSL_PROVIDER_load(libctx, modulename);
+    if (!TEST_ptr(prov)
+               /* Check we have the provider available */
+            || !TEST_true(OSSL_PROVIDER_available(libctx, modulename))
+               /* Check the default provider is not available */)
```

Comment:
Comment should be below the goto shouldn't it?

Reply:
Oops. An artefact from an earlier code iteration. Fixed now.